spring-boot/proxy/src/test/java/eu/coatrack/proxy/security/local_api_key_manager_tests/ApiKeyFindingInLocalApiKeyListTest.java

```diff
@@ -0,0 +1,105 @@
+package eu.coatrack.proxy.security.local_api_key_manager_tests;/*-
+ * #%L
+ * coatrack-proxy
+ * %%
+ * Copyright (C) 2013 - 2021 Corizon | Institut für angewandte Systemtechnik Bremen GmbH (ATB)
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import eu.coatrack.api.ApiKey;
+import eu.coatrack.proxy.security.exceptions.ApiKeyFetchingFailedException;
+import eu.coatrack.proxy.security.LocalApiKeyManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+public class ApiKeyFindingInLocalApiKeyListTest extends AbstractLocalApiKeyManagerSetup {
+
+    @BeforeEach
+    public void fillLocalApiKeyListWithListContainingValidApiKey() throws ApiKeyFetchingFailedException {
+        super.setupLocalApiKeyManagerAndApiKeyList();
+        when(apiKeyFetcherMock.requestLatestApiKeyListFromAdmin()).thenReturn(apiKeyList);
+        localApiKeyManager.updateLocalApiKeyList();
+    }
+
+    @Test
+    public void nullApiKeyValueShouldBeAnsweredWithNull() {
+        assertNull(localApiKeyManager.getApiKeyEntityByApiKeyValue(null));
```

Comment:
Just an idea: Should this maybe throw an exception instead of returning null?

Reply:
I am uncertain about what would be best practice here. I asserted on a higher level that apiKeyValue is not null:
```Assert.hasText(apiKeyValue, "The credentials did not contain any letters.");```

There are two approaches in my opinion:
* Leaving out null checks, exception handling and null-tests on low level methods because apiKeyValue can't be null.
* Create a null check (with IllegalArgumentException) on every lower level method. This would cause more code but in case of refactoring it is assured that if a null-related bug occurs it can be found quickly.

What do you think about that?

Reply:
If an incoming call to a CoatRack gateway does not contain any api key value, the whole authentication process does not make any sense. So I agree to do a null check at the very beginning, and if it is null, we will not need to "bother" the `LocalApiKeyManager` or the `ApiKeyFetcher`. Therefore, the case in this test will only become relevant in case of errors in other parts of our code, this is why maybe an exception makes sense to find the error early. But you can also leave it as it is, because in normal operation it should not happen.

Reply:
Good point. Thanks for the explanation.